SQL/SQL (basic).sublime-syntax

```diff
@@ -0,0 +1,890 @@
+      scope: keyword.other.ddl.sql
+      push:
+        - ddl-alter-table
+        - expect-table-name
+    - match: \b(?i:alter)\b
+      scope: keyword.other.ddl.sql
+      push:
+        - ddl-alter-target
+        - ddl-target
+
+  ddl-create-target:
+    - meta_scope: meta.create.sql
+    - include: ddl-target-on
+
+  ddl-create-target-expect-as:
+    - match: \b(?i:as)\b
+      scope: keyword.context.block.sql
+      pop: 1
+    - match: \b(?i:returns)\b
+      scope: keyword.other.sql
+      push: expect-type
+    - match: (@){{simple_identifier}}
+      scope: variable.parameter.sql
+      push: expect-type
+    - include: expressions
+    - include: pop-on-top-level-reserved-word
+
+  ddl-target-on:
+    - match: \b(?i:on)\b
+      scope: keyword.other.sql
+      push: expect-table-name
+    - include: else-pop
+
+  ddl-drop-target:
+    - meta_include_prototype: false
+    - meta_scope: meta.drop.sql
+    - include: immediately-pop
+
+  ddl-target:
+    - match: \b{{ddl_target}}\b
+      scope: keyword.other.sql
+    - include: else-pop
+
+  ddl-drop-table:
+    - meta_scope: meta.drop.sql
+    - include: dml-condition
+    - include: expect-table-name
+
+  ddl-alter-table:
+    - meta_scope: meta.alter.sql
+    - include: ddl-alter-column
+    - include: ddl-alter-common
+    - include: expressions-or-column-names
+
+  ddl-alter-target:
+    - meta_scope: meta.alter.sql
+    - include: ddl-alter-common
+    - include: else-pop
+
+  ddl-alter-common:
+    - match: \b(?i:add\s+constraint)\b
+      scope: meta.add.sql keyword.other.sql
+      push:
+        - maybe-column-modifier
+        - expect-constraint-name
+    - match: \b(?i:add\s+(?:(?:fulltext|spatial)\s+(index|key)|index))\b
+      scope: meta.add.sql keyword.other.sql
+    - include: pop-on-top-level-reserved-word
+
+  ddl-alter-column:
+    - match: \b(?i:(?:add|alter)\s+column)\b
+      scope: keyword.other.ddl.sql
+      push:
+        - expect-type
+        - expect-column-name-declaration
+    - match: \b(?i:(?:add|alter))\b(?!\s+{{ddl_target}}\b)
+      scope: keyword.other.ddl.sql
+      push:
+        - expect-type
+        - expect-column-name-declaration
+    - match: \b(?i:drop\s+column)\b
+      scope: keyword.other.ddl.sql
+      push: expect-column-name
+    - match: \b(?i:drop)\b(?!\s+{{ddl_target}}\b)
+      scope: keyword.other.ddl.sql
+      push: expect-column-name
+
+  create-other-condition:
+    - include: dml-condition
+    - include: expect-other-creation-name
+
+  create-table-condition:
+    - include: dml-condition
+    - include: expect-table-creation-name
+
+  drop-condition:
+    - include: dml-condition
+    - match: (?=\S)
+      set:
+        - ddl-target-on
+        - single-identifier
+
+###[ DML STATEMENTS ]##########################################################
+
+  dml-statements:
+    - match: \b(?i:select)\b
+      scope: keyword.other.dml.sql
+    - match: \b(?i:union(?:\s+all)?)\b
+      scope: keyword.other.dml.sql
```

Comment:
`keyword.operator.set`? 😜

Reply:
My mistake. `keyword.operator.logical`

Reply:
The whole scope naming of various statements/expressions is probably worth some refactoring. Historically many keywords are just scoped `keyword.other.dml` or `keyword.other.ddl` which seems rather over used.

It's sometimes not even very clear whether a keyword is a keyword or a constant or even some kind of function being called.

Finding good scope names may however require to finish implementation of main supported dialects first to get a better overview about what's needed. With each statement requireing its own syntax definition tree, this is a rather comprehensive task though.